### DISCUSSION TITLE
How to use hls4ml[main]

### DISCUSSION THREAD

**Discussion by zyt1024** [fastmachinelearning/hls4ml#822] [General]
Hello，Now, I have installed hls4ml 0.7.1 according to the tutorial, but I want to use the hls4ml library under the main branch of hls4ml. How can I install and replace hls4ml 0.7.1 in a virtual environment?

**Comment by jmduarte** [comment]
You can install via pip with the profiling extras
```bash
pip install git+https://github.com/fastmachinelearning/hls4ml.git@main#egg=hls4ml[profiling]
```
See https://adamj.eu/tech/2019/03/11/pip-install-from-a-git-repository/

Be warned it is better to do this at a particular commit so you know how to reproduce the environment.

**Reply by zyt1024** [reply]
Ok,Thanks,and I will try it.
